preparedThreshold=0 ignored when doing DatabaseMetaData calls

**Describe the issue**
* The driver results in an error such as `prepared statement "S_X" already exists` when reading Metadata for a `PreparedStatement` when using PGBouncer
* The JDBC URL contains `preparedThreshold=0` to stop server side prepared statements
* The behaviour is inconsistent, sometimes running the same bit of code works 1 or 2 times but then the error occurs. However, running the code again an hour or so later, the code then works 1 or 2 times again, and so on.
* Looking at the noteable changes here: https://jdbc.postgresql.org/changelogs/2025-09-18-42/ in version `42.7.8` of the driver, you can see the link `perf: enable server-prepared statements for DatabaseMetaData`
* This behaviour did NOT occur in the previous version of the driver (`42.7.5`) so, given the release note above, it suggests possibly a new bug has been introduced
* A work around is to explicitly set `preferQueryMode=simple` in the properties too but my understanding is `preparedThreshold=0` should be enough?
* Another observation is that actual JDBC calls (e.g. SELECTs, INSERTs, etc.) seem to be okay; it's just when getting `DatabaseMetaData`

**Driver Version?** 
42.7.8

**Java Version?**
21

**PostgreSQL Version?**
`PostgreSQL 14.19 on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0, 64-bit`

**To Reproduce**

1. Create a `PreparedStatement` using the `javax.sql.Datasource` connection via `connection.prepareStatement(sql)`
2. Get the `ResultSetMetaData` of that `PreparedStatement` using `statement.getMetaData()`

With the following circumstances 
1. The `javax.sql.Datasource` is an instance of `HikariProxyConnection`; which delegates to `PgConnection`
2. PGBouncer is used with `pool_mode=transaction`
3. Spring Boot version is `3.4.12`
4. The JDBC URL has `preparedThreshold=0`

**Expected behaviour**
Server Prepared Statements **not** to be used on the call to `getMetaData`
Instead, intermittently, an error such as `prepared statement "S_X" already exists` is returned

**Logs**
```
Caused by: org.postgresql.util.PSQLException: ERROR: prepared statement "S_4" already exists
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2736)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2421)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:330)
	at org.postgresql.jdbc.PgPreparedStatement.getMetaData(PgPreparedStatement.java:1207)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.getMetaData(HikariProxyPreparedStatement.java)
```

--- a/pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java
+++ b/pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java
@@ -1204,6 +1204,9 @@ class PgPreparedStatement extends PgStatement implements PreparedStatement {
 
       int flags = QueryExecutor.QUERY_DESCRIBE_ONLY
           | QueryExecutor.QUERY_SUPPRESS_BEGIN;
+      if (isOneShotQuery(null)) {
+        flags |= QueryExecutor.QUERY_ONESHOT;
+      }
       StatementResultHandler handler = new StatementResultHandler();
       connection.getQueryExecutor().execute(preparedQuery.query, preparedParameters, handler, 0, 0,
           flags);

--- a/pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java
+++ b/pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java
@@ -1736,6 +1736,40 @@ public class PreparedStatementTest extends BaseTest4 {
   }
 
   @Test
+  public void testGetMetaDataWithPrepareThreshold0() throws SQLException {
+    assumeBinaryModeRegular();
+    assumeTrue(preferQueryMode != PreferQueryMode.SIMPLE, "simple protocol only does not support prepared statement requests");
+
+    PreparedStatement pstmt = null;
+    try {
+      pstmt = con.prepareStatement("SELECT 43");
+      ((PgStatement) pstmt).setPrepareThreshold(0);
+      assertNotNull(pstmt.getMetaData(), "getMetaData() should describe the statement");
+    } finally {
+      TestUtil.closeQuietly(pstmt);
+    }
+
+    assertEquals(0, getNumberOfServerPreparedStatements("SELECT 43"), "prepareThreshold=0, so getMetaData() should not server-prepare the statement");
+  }
+
+  @Test
+  public void testGetMetaDataDoesNotBypassPrepareThreshold() throws SQLException {
+    assumeBinaryModeRegular();
+    assumeTrue(preferQueryMode != PreferQueryMode.SIMPLE, "simple protocol only does not support prepared statement requests");
+
+    PreparedStatement pstmt = null;
+    try {
+      pstmt = con.prepareStatement("SELECT 44");
+      ((PgStatement) pstmt).setPrepareThreshold(5);
+      assertNotNull(pstmt.getMetaData(), "getMetaData() should describe the statement");
+    } finally {
+      TestUtil.closeQuietly(pstmt);
+    }
+
+    assertEquals(0, getNumberOfServerPreparedStatements("SELECT 44"), "prepareThreshold=5 and the statement was never executed, so describing it should not server-prepare it");
+  }
+
+  @Test
   public void testInappropriateStatementSharing() throws SQLException {
     PreparedStatement ps = con.prepareStatement("SELECT ?::timestamp");
     assertFirstParameterTypeName("after prepare ?::timestamp bind type should be timestamp", "timestamp", ps);

